lndclient/lnd_services.go

```diff
@@ -198,28 +247,146 @@ func (s *GrpcLndServices) Close() {
 }
 
 // checkLndCompatibility makes sure the connected lnd instance is running on the
-// correct network.
+// correct network, has the version RPC implemented, is the correct minimal
+// version and supports all required build tags/subservers.
 func checkLndCompatibility(conn *grpc.ClientConn, chainParams *chaincfg.Params,
-	readonlyMac serializedMacaroon, network string) error {
+	readonlyMac serializedMacaroon, network string,
+	minVersion *verrpc.Version) (string, [33]byte, *verrpc.Version, error) {
+
+	// onErr is a closure that simplifies returning multiple values in the
+	// error case.
+	onErr := func(err error) (string, [33]byte, *verrpc.Version, error) {
+		closeErr := conn.Close()
+		if closeErr != nil {
+			log.Errorf("Error closing client connection: %v",
+				closeErr)
+		}
 
-	lightningClient := newLightningClient(
-		conn, chainParams, readonlyMac,
-	)
+		return "", [33]byte{}, nil, fmt.Errorf("error checking "+
+			"connected lnd version. at least version \"%s\" is "+
+			"required: %v", VersionString(minVersion), err)
+	}
+
+	// We use our own clients with a readonly macaroon here, because we know
+	// that's all we need for the checks.
+	lightningClient := newLightningClient(conn, chainParams, readonlyMac)
+	versionerClient := newVersionerClient(conn, readonlyMac)
 
 	// With our readonly macaroon obtained, we'll ensure that the network
 	// for lnd matches our expected network.
 	info, err := lightningClient.GetInfo(context.Background())
 	if err != nil {
-		conn.Close()
-		return fmt.Errorf("unable to get info for lnd "+
-			"node: %v", err)
+		return onErr(fmt.Errorf(
+			"unable to get info for lnd node: %v", err,
+		))
 	}
 	if network != info.Network {
-		conn.Close()
-		return errors.New(
-			"network mismatch with connected lnd instance",
-		)
+		return onErr(fmt.Errorf(
+			"network mismatch with connected lnd instance. "+
+				"wanted '%s', got '%s'", network, info.Network,
+		))
+	}
+
+	// Now let's also check the version of the connected lnd node.
+	err = checkVersionCompatibility(versionerClient, minVersion)
```

Comment:
`GetVersion` is called twice now. Could also pass the retrieved version into `checkVersionCompatibility`